### ISSUE TITLE
json-output-folder not cleaned

### ISSUE THREAD

**Issue by robbinvandenberg** [wswebcreation/protractor-multiple-cucumber-html-reporter-plugin#24]
**Environment (please complete the following information):**
 - protractor-multiple-cucumber-html-reporter: 1.8.0
 - Node.js version: 8.12.0
 - NPM version: 6.4.1
 - Platform name and version: Windows 10
 - protractor version: 5.4.2
 - Cucucmber version: 5.1.0
 - protractor-cucumber-framework version: 6.1.1

**Config of protractor + protractor-multiple-cucumber-html-reporter**

```
exports.config = {
  /**
   * A base URL for your application under test. Calls to protractor.get()
   * with relative paths will be resolved against this URL (via url.resolve)
   */
  baseUrl: 'http://localhost:4200/',

  // set to "custom" instead of cucumber.
  framework: 'custom',

  // path relative to the current config file
  frameworkPath: require.resolve('protractor-cucumber-framework'),

  // require feature files
  specs: [
    './src/features/*.feature' // accepts a glob
  ],

  cucumberOpts: {
    // require step definitions
    require: [
      './src/steps/*.e2e-spec.ts', // accepts a glob
      './src/support/*.ts' // Include support files (hooks.ts)
    ],
    // Only execute scenarios with tags below
    tags: '@automateds',
    // Format console output when running feature tests
    //format: ['snippets'],
    format: ['node_modules/cucumber-pretty', 'json:.tmp/results.json'],
    'fail-fast': true
  },

  // Enable TypeScript for the tests
  onPrepare() {
    require('ts-node').register({
      project: 'e2e/tsconfig.e2e.json'
    });
    // Maximize browser window
    browser.manage().window().maximize();
  },
  // If your process abruptly stops with an exit code 199 then your steps most likely threw an uncaught exception. 
  // Protractor is capturing these and exiting the process in this situation. 
  // The line below allows cucumber to handle the exception and record it appropriately.
  ignoreUncaughtExceptions: false,
  // Set in which browsers to execute the tests
  multiCapabilities: [
    { 'browserName': 'chrome' },
  ],
  // Run tests in series instead of parallel
  maxSessions: 1,
  // Protractor waits for the angular variable to be present when loading a new page, set the timeout for this wait in ms (default 10 seconds)
  getPageTimeout: 120000,
  // The timeout in milliseconds for each script run on the browser. This should be longer than the maximum time your application needs to stabilize between tasks.
  allScriptsTimeout: 120000,

  // Setup generation of feature test report plugin
  plugins: [{
    package: 'protractor-multiple-cucumber-html-reporter-plugin',
    options:{
      automaticallyGenerateReport: true,
      removeExistingJsonReportFile: true,
      removeOriginalJsonReportFile: true,
      displayDuration: true
    }
  }]
};
```

**Describe the bug**
When executing for example two tests runs with a different amount of feature files, legacy outputs of the previous test are shown in the html report.

**To Reproduce**

- Run a test with 2 feature files
- Automated generated HTML report contains the correct data of the 2 feature files
- Run another test but with 1 feature file
- Automated generated HTML report also contains the data of the feature files from the first test run, which is not what I expect.

**Expected behavior**
I expect that the output of the generated html report is not affected by independent previous runs, and only contains data about the last test run.

**Additional context**
It seems like the json-output-folder is not cleaned before a new test run.
I also tried expirementing with the removeExistingJsonReportFile and removeOriginalJsonReportFile options, but that didn't resolve the issue

**Comment by MasterGru** [comment]
hi, i have the same issue.
@Master
could you update the package of multiple-cucumber-html-reporter ? because the #55 is not included in the 1.8.0 version of protractor-multiple-cucumber-html-report

thanks,
Leo

**Comment by wswebcreation** [comment]
@robbinvandenberg and @MasterGru 

Cleaning the folder which holds data of previous runs is not possible. The architecture of Protractor Plugins is that it will will be called for each instance (read capability) in your config. This means that if I would provide an option to remove data from previous runs, it might remove data from an instance that was just finished running.

There is no `beforeLaunch` and `onComplete` hook available within the plugin architecture that might be able to do that

The `removeExistingJsonReportFile` does the following, see [this](https://github.com/wswebcreation/protractor-multiple-cucumber-html-reporter-plugin#removeexistingjsonreportfile).

Going to close this issue because it can't be done from a plugin perspective, you can do something in your own config file by adding this in the `exports.config = {`

```js
// Install 'fs-extra' as a dev-dependency
const fs = require('fs-extra');

exports.config = {
    // ....
    beforeLaunch() => {
        fs.removeSync('your-folder-that-holds-the-files');
    }
}
```
